ton_client/src/debot/calltype.rs

```diff
@@ -161,8 +167,14 @@ async fn decode_and_fix_ext_msg(
 
     let mut new_body = BuilderData::new();
     let pubkey = signer.resolve_public_key(ton.clone()).await?;
-    if meta.is_pubkey {
-        if let Some(ref key) = pubkey {
+    if meta.is_pubkey ||  meta.ex_provided_pubkey.is_some() {
+        if let Some(ref ex_key) = meta.ex_provided_pubkey {
+            new_body
```

Comment:
Can you make a little optimization here? Choose pubkey first and then append to the body.
Like the following:
```
let appending_key = match meta.ex_provided_pubkey {
    Some(ref ex_key) => ex_key,
    None => &key,
}
new_body .append_bit_one()
                .and_then(|b| b.append_raw(&hex::decode(appending_key).unwrap(), 256))
                .map_err(msg_err)?;
```